common/ini-file.cpp

```diff
@@ -28,11 +28,20 @@
 namespace Common {
 
 bool INIFile::isValidName(const String &name) const {
-	if (_allowNonEnglishCharacters)
-		return true;
+	if (name.empty())
+		return false;
+
 	const char *p = name.c_str();
-	while (*p && (isAlnum(*p) || *p == '-' || *p == '_' || *p == '.' || *p == ' ' || *p == ':'))
-		p++;
+	if (_allowNonEnglishCharacters) {
+		// Only disallow characters which have meaning in ini files
+		// and could break parsing
+		while (*p && *p != '[' && *p != ']' && *p != '=' && *p != '#' && *p != '\r' && *p != '\n')
```

Comment:
Could this have a similar logic with the branch below (where non-English characters aren't allowed)?
With this change, handling of allowed characters is quite different between the two logic branches

Reply:
I tried combining into one branch but couldn't see a nice way to do it.  Open to suggestions though.

Previously in `allowNonEnglishCharacters` mode it allowed *any* characters, even ones that would break parsing.  This is trying to only add a minimal list of disallowed characters, so by design the list is very different from the "allowed" characters in the other mode.

Reply:
Thinking about it more, it might be nice to add some small statics like `alwaysAllowedChar()` and `neverAllowedChar()`, then I can simplify the loop.  I'll do that.

Reply:
Had a busy week and only just got back to this.  I decided it was cleaner to just add `isValidChar` and I think it's nicer now, let me know what you think.